shell_integration/dolphin/ownclouddolphinpluginhelper.cpp

```diff
@@ -62,18 +62,17 @@ void OwncloudDolphinPluginHelper::sendCommand(const char* data)
 
 void OwncloudDolphinPluginHelper::sendGetClientIconCommand(int size)
 {
-    QByteArray line_end("\n");
-    QJsonObject args { { "size", size } };
-    QJsonObject obj { { QStringLiteral("id"), "1" }, { QStringLiteral("arguments"), args } };
-    std::string command = "V2/GET_CLIENT_ICON:" + QJsonDocument(obj).toJson(QJsonDocument::Compact).toStdString();
-    sendCommand(QByteArray(command.c_str() + line_end));
+    int msgId = 1;
+    QJsonObject args { { QStringLiteral("size"), size } };
+    QJsonObject obj { { QStringLiteral("id"), QString(msgId) }, { QStringLiteral("arguments"), args } };
+    auto json = QJsonDocument(obj).toJson(QJsonDocument::Compact);
+    sendCommand(QByteArray("V2/GET_CLIENT_ICON:" + json + "\n"));
```

Comment:
For all the mess with the strings, we build our core library with a stricter string handling, so for example QString("a") would not compile there. That's why I'm a bit pedantic here :)

QByteArray("V2/GET_CLIENT_ICON:" + json + "\n")

Reply:
improved - hopefully correct now